README.md

```diff
@@ -101,12 +101,15 @@ This library supports reading all Avro types, with the exception of complex `uni
 | array     | ArrayType      |
 | map       | MapType        |
 | fixed     | BinaryType     |
+| union     | See below      |
 
-In addition to the types listed above, it supports reading of three types of `union` types:
+In addition to the types listed above, it supports reading `union` types. The following three types are considered basic `union` types:
 
-1. `union(int, long)`
-2. `union(float, double)`
-3. `union(something, null)`, where `something` is one of the supported Avro types listed above or is one of the supported `union` types.
+1. `union(int, long)` will be mapped to `LongType`.
+2. `union(float, double)` will be mapped to `DoubleType`.
+3. `union(something, null)`, where `something` is any supported Avro type. This will be mapped to the same Spark SQL type as that of `something`, with `nullable` set to `true`.
```

Comment:
This makes me curious about the `union(union(int, long), null)` case. Let me quickly double-check to make sure that there's a test covering this corner-case to make sure that behavior hasn't changed here.